operations/mimir-mixin/dashboards/writes.libsonnet

```diff
@@ -206,6 +235,232 @@ local filename = 'mimir-writes.json';
+            'histogram_quantile(0.99, sum by(le) (rate(cortex_ingest_storage_reader_records_per_fetch_bucket{%s}[$__rate_interval])))' % [$.jobMatcher($._config.job_names.ingester)],
+          ],
+          [
+            'Average',
+            '99th percentile',
+          ],
+        ),
+      )
+      .addPanel(
+        $.timeseriesPanel('Processing Latency') +
+        $.panelDescription(
+          'Processing Latency',
+          |||
+            Time used to process a single record (write request). This time is spent by appending data to per-tenant TSDB.
+          |||
+        ) +
+        $.queryPanel(
+          [
+            'max(cortex_ingest_storage_reader_processing_time_seconds{%s,quantile="0.99"})' % [$.jobMatcher($._config.job_names.ingester)],
+            'max(cortex_ingest_storage_reader_processing_time_seconds{%s,quantile="0.999"})' % [$.jobMatcher($._config.job_names.ingester)],
+            'max(cortex_ingest_storage_reader_processing_time_seconds{%s,quantile="0.5"})' % [$.jobMatcher($._config.job_names.ingester)],
+            |||
+              sum(rate(cortex_ingest_storage_reader_processing_time_seconds_sum{%s}[$__rate_interval]))
+              /
+              sum(rate(cortex_ingest_storage_reader_processing_time_seconds_count{%s}[$__rate_interval]))
+            ||| % [$.jobMatcher($._config.job_names.ingester), $.jobMatcher($._config.job_names.ingester)],
+          ],
+          [
+            '99th percentile',
+            '99.9th percentile',
+            '50th percentile',
+            'average',
+          ],
+        ) + {
+          fieldConfig+: {
+            defaults+: { unit: 's' },
+          },
+        },
+      )
+      .addPanel(
+        $.timeseriesPanel('End-to-end latency') +
+        $.panelDescription(
+          'End-to-end latency',
+          |||
+            Time between writing request to Kafka by distributor and reading the record by ingester.
+          |||
+        ) +
+        $.queryPanel(
+          [
+            'max(cortex_ingest_storage_reader_receive_delay_seconds{%s,quantile="0.99"})' % [$.jobMatcher($._config.job_names.ingester)],
+            'max(cortex_ingest_storage_reader_receive_delay_seconds{%s,quantile="0.999"})' % [$.jobMatcher($._config.job_names.ingester)],
+            'max(cortex_ingest_storage_reader_receive_delay_seconds{%s,quantile="0.5"})' % [$.jobMatcher($._config.job_names.ingester)],
+          ],
+          [
+            '99th percentile',
+            '99.9th percentile',
+            '50th percentile',
+          ],
+        ) + {
+          fieldConfig+: {
+            defaults+: { unit: 's' },
+          },
+        },
+      )
+    )
+    .addRowIf(
+      $._config.show_ingest_storage_panels,
+      ($.row('Ingester (ingest storage: strong consistency)'))
+      .addPanel(
+        $.timeseriesPanel('Requests with strong consistency') +
+        $.panelDescription(
+          'Requests with strong consistency',
```

Comment:
What do you think if we transform this into "Strong consistency queries ratio" (in the Queries dashboard we have "Sharded queries ratio" which is similar concept, even if tracked by query-frontend). It would show the % of strong consistency requests over the total. We could have both the % of succeeded and failed requests.